### PR TITLE
Add cases for subcommand rebase of qemu-img

### PR DESCRIPTION
**Pull request by pingl2016**
Add cases for cache mode and show progress

Signed-off-by: Ping Li pingl@redhat.com

ID: 1361861, 1362614

### CODE REVIEW

File: qemu/tests/cfg/qemu_img.cfg [hunk 1]
```diff
@@ -90,9 +90,20 @@
         - rebase:
             no raw vmdk qed
             subcommand = rebase
-            rebase_mode = unsafe
-            image_name_snapshot1 = sn1
-            image_name_snapshot2 = sn2
+            image_name_snapshot1 = "images/sn1"
+            image_name_snapshot2 = "images/sn2"
+            variants:
+                -unsafe:
```
Comment (xiagao) [inline qemu/tests/cfg/qemu_img.cfg]: leave blank here pls.

**Comment by PyLearner** [comment]
Acked-by: MengYang meyang@redhat.com

 (1/5) smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_img.convert.to_qcow2: PASS (258.62 s)
 (2/5) smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_img.convert.to_qcow2.cluster_size_2048: PASS (246.31 s)
 (3/5) smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_img.convert.to_qcow2.show_progress.show_progress_on: PASS (132.53 s)
 (4/5) smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_img.convert.to_qcow2.show_progress.show_progress_off: PASS (889.91 s)
 (5/5) smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_img.convert.to_raw: PASS (134.60 s)


 (1/3) smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_img.rebase.unsafe: PASS (40.74 s)
 (2/3) smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_img.rebase.show_progress.show_progress_on: PASS (170.52 s)
 (3/3) smp_16.16384m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_img.rebase.show_progress.show_progress_off: PASS (169.69 s)

File: qemu/tests/qemu_img.py [hunk 2]
```diff
@@ -520,7 +523,7 @@ def rebase_test(cmd):
         sn2 = _get_image_filename(sn2, enable_gluster, sn_fmt)
         _create(cmd, sn2, sn_fmt, base_img=sn1, base_img_fmt=sn_fmt)
 
-        rebase_mode = params.get("rebase_mode")
+        rebase_mode = params.get("rebase_mode", "safe")
```
Comment (suqinhuang) [inline qemu/tests/qemu_img.py]: "safe" is the default mode, right?

File: qemu/tests/qemu_disk_img_rebase.py [hunk 3]
```diff
@@ -68,51 +68,68 @@ def run(test, params, env):
...
-                raise error.TestError("Check md5sum fail (file:%s)" % _file)
-        rebase_test.destroy_vm()
-        rebase_test.check_image()
+            rebase_test = RebaseTest(test, params, env, image)
+            t_params = rebase_test.rebase()
+            rebase_test.check_backingfile()
+            rebase_test.start_vm(t_params)
+            check_files = params.get("check_files", "").split()
+            for _file in check_files:
+                ret = rebase_test.check_file(_file, md5_dict[_file])
```
Comment (suqinhuang) [inline qemu/tests/qemu_disk_img_rebase.py]: Could you double check the step description and make sure it's added for every step

Comment (pingl2016) [inline qemu/tests/qemu_disk_img_rebase.py]: the description of primary steps have been added, such as create snapshot, start guest, save temp file in guest, do the rebase test and re-check the md5 of temp file in the guest.

**Comment by xiywang** [comment]
Acked-by: Xiyue Wang xiywang@redhat.com

 (1/3) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.myprovider.qemu_disk_img.rebase.snB.to_base: PASS (293.66 s)
 (2/3) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.myprovider.qemu_disk_img.rebase.snB.to_base.cache_mode_all: PASS (1503.08 s)
 (3/3) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.myprovider.qemu_disk_img.rebase.snB.to_null: PASS (404.77 s)

 (1/3) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.myprovider.qemu_img.rebase.unsafe: PASS (55.12 s)
 (2/3) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.myprovider.qemu_img.rebase.show_progress.show_progress_on: PASS (115.53 s)
 (3/3) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.myprovider.qemu_img.rebase.show_progress.show_progress_off: PASS (115.40 s)

File: qemu/tests/qemu_disk_img_rebase.py [hunk 4]
```diff
@@ -68,51 +68,68 @@ def run(test, params, env):
...
-        t_file = params["guest_file_name_%s" % tag]
-        md5 = rebase_test.save_file(t_file)
-        if not md5:
-            raise error.TestError("Fail to save tmp file")
-        md5_dict[t_file] = md5
-        rebase_test.destroy_vm()
+    values = ""
+    keys = params.keys()
+    keystr = (' ').join(keys)
+    key = re.findall("rebase_list_(\S+)", keystr)
```
Comment (suqinhuang) [inline qemu/tests/qemu_disk_img_rebase.py]: not clear when case failed

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

**Pull request merged**